Python/Product/Attacher/PyAttacher.cs

```diff
@@ -466,12 +470,17 @@ internal sealed class DebugAttach : IDisposable {
                             viewStream.Write(BitConverter.GetBytes(attachDoneTargetHandle.ToInt64()), 0, 8);
 
                             var errorCodePosition = viewStream.Position;
-
                             viewStream.Write(new byte[8], 0, 8); // write null bytes for error code and version
-                            string guid = debugId.ToString();
-                            for (int i = 0; i < guid.Length; i++) {
-                                viewStream.WriteByte((byte)guid[i]);
+
+                            string szDebugId = debugId.ToString().PadRight(64, '\0');
+                            for (int i = 0; i < szDebugId.Length; i++) {
+                                viewStream.WriteByte((byte)szDebugId[i]);
+                            }
+
+                            for (int i = 0; i < debugOptions.Length; i++) {
+                                viewStream.WriteByte((byte)debugOptions[i]);

```

Comment:
I'm going to use Encoding.ASCII here, because in practice that is what this is anyway, and using UTF8 will make length validation when parsing command line arguments more complicated.